dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -237,106 +237,122 @@ default List<SourceFile> getAllSourceFiles(long workflowId, String tag, Descript
             .map(entry -> entry.getValue().getLeft()).collect(Collectors.toList());
     }
 
+    default String getDefaultVersionName() {
+        return getDAO() instanceof ToolDAO ? "latest" : "master";
+    }
+
+    /**
+     * Finds the version and entry corresponding to the specified version name and entry ID, adjusting access per the specified logged-in user.
+     * The entry will only contain the version with the specified version name, and will be evicted so no accidental db mods are possible.
+     * Throws a CustomWebApplicationException if the entry or version is not accessible.
+     * @param entryId id of the entry
+     * @param versionName name of the version
+     * @param user the logged-in user
+     * @return the version and parent entry
+     */
+    default EvictedVersionAndEntry<T> findAndCheckVersionByName(long entryId, String versionName, Optional<User> user, VersionDAO versionDAO) {
+        try {
+            String modifiedVersionName = ObjectUtils.firstNonNull(versionName, getDefaultVersionName());
+            versionDAO.enableNameFilter(modifiedVersionName);
+            T entry = findAndCheckEntryById(entryId, user);
+            Version version = entry.getWorkflowVersions().stream().filter(v -> v.getName().equals(modifiedVersionName)).findFirst().orElseThrow(
+                () -> new CustomWebApplicationException("Invalid or missing version " + modifiedVersionName + ".", HttpStatus.SC_BAD_REQUEST));
```

Comment:
The version could be hidden too, but I think this message is OK in that case -- from the end-user perspective, we should treat hidden versions as missing.